libs/barista-components/experimental/quick-filter/src/quick-filter.ts

```diff
@@ -168,13 +168,22 @@ export class DtQuickFilter<T = any> implements AfterViewInit, OnDestroy {
 
   /** Angular life-cycle hook that will be called after the view is initialized */
   ngAfterViewInit(): void {
-    // When the filters changes apply them to the filter field
-    this._activeFilters$
-      .pipe(takeUntil(this._destroy$))
-      .subscribe((filters) => {
-        if (this._filterField.filters !== filters) {
-          this._filterField.filters = filters;
-        }
+    // We need to wait for the first on stable call, otherwise the
+    // underlying filterfield will thow an expression changed after checked
+    // error. Deferring the first filter setting.
+    // Relates to a very weird and hard to reproduce bug described in
+    // https://github.com/dynatrace-oss/barista/issues/1305
+    this._zone.onStable
+      .pipe(take(1))
+      // When the filters changes apply them to the filter field
+      .subscribe(() => {
+        this._activeFilters$
+          .pipe(takeUntil(this._destroy$))
+          .subscribe((filters) => {
+            if (this._filterField.filters !== filters) {
+              this._filterField.filters = filters;
+            }
+          });
       });
```

Comment:
```suggestion
    this._zone.onStable
      .pipe(
        take(1),
        switchMap(() => this._activeFilters$),
        takeUntil(this._destroy$)      
    ).subscribe((filters) => {
            if (this._filterField.filters !== filters) {
              this._filterField.filters = filters;
            }
          });
      });
```

Reply:
This should work out the same and avoids the nested subscriptions. 

https://stackblitz.com/edit/rxjs-ck7pba?devtoolsheight=60

Reply:
To my understanding, this would kill the subscription after the first on stable (because of the take(1)). 

I created the nested subscription, because I only want to subscribe after the first onStable, but want to keep the inner subscription around for the duration of the component lifetime. 

Or am I completely missing how rxjs switchmap and unsubscriptions work?

Reply:
Apparently I did blank on this one. Updated to your suggestion in the code.